examples/simulation/plot_stc_metrics.py

```diff
@@ -0,0 +1,286 @@
+mne.simulation.add_noise(raw_region, cov, iir_filter=[0.2, -0.2, 0.04],
+                         random_state=random_state)
+
+# Dipole
+raw_dipole = mne.simulation.simulate_raw(info, source_simulator_dipole,
+                                         forward=fwd)
+raw_dipole = raw_dipole.pick_types(meg=False, eeg=True, stim=True)
+cov = mne.make_ad_hoc_cov(raw_dipole.info)
+mne.simulation.add_noise(raw_dipole, cov, iir_filter=[0.2, -0.2, 0.04],
+                         random_state=random_state)
+
+###############################################################################
+# Compute evoked from raw data
+# ----------------------------
+#
+# Averaging epochs corresponding to events.
+
+# Region
+events = mne.find_events(raw_region, initial_event=True)
+tmax = (len(source_time_series) - 1) * tstep
+epochs = mne.Epochs(raw_region, events, 1, tmin=0, tmax=tmax)
+evoked_region = epochs.average()
+
+# Dipole
+events = mne.find_events(raw_dipole, initial_event=True)
+tmax = (len(source_time_series) - 1) * tstep
+epochs = mne.Epochs(raw_dipole, events, 1, tmin=0, tmax=tmax)
+evoked_dipole = epochs.average()
+
+###############################################################################
+# Create true stcs corresponding to evoked
+# ----------------------------------------
+#
+# Before we computed stcs corresponding to raw data. To be able to compare
+# it with the reconstruction, based on the evoked, true stc should have the
+# same number of time samples.
+
+# Region
+stc_true_region = \
+    source_simulator_region.get_stc(start_sample=0,
+                                    stop_sample=len(source_time_series))
+
+# Dipole
+stc_true_dipole = \
+    source_simulator_dipole.get_stc(start_sample=0,
+                                    stop_sample=len(source_time_series))
+
+###############################################################################
+# Reconstruct simulated sources
+# -----------------------------
+#
+# Compute inverse solution using sLORETA.
+
+# Region
+snr = 30.0
+inv_method = 'sLORETA'
+lambda2 = 1.0 / snr ** 2
+
+inverse_operator = make_inverse_operator(evoked_region.info, fwd, cov,
+                                         loose='auto', depth=0.8,
+                                         fixed=True)
+
+stc_est_region = apply_inverse(evoked_region, inverse_operator, lambda2,
+                               inv_method, pick_ori=None)
+
+# Dipole
+snr = 3.0
+inv_method = 'sLORETA'
+lambda2 = 1.0 / snr ** 2
+
+inverse_operator = make_inverse_operator(evoked_dipole.info, fwd, cov,
+                                         loose='auto', depth=0.8,
+                                         fixed=True)
+
+stc_est_dipole = apply_inverse(evoked_dipole, inverse_operator, lambda2,
+                               inv_method, pick_ori=None)
+
+###############################################################################
+# Compute performance scores for different source amplitude thresholds
```

Comment:
Do not handcraft the analysis, plotting etc. Use sklearn/pandas utilites.

See [sleep example](https://mne-tools.github.io/dev/auto_tutorials/sample-datasets/plot_sleep.html?highlight=sleep)

Reply:
what do you have in mind @massich ?

Reply:
I can either make a PR or push here. I'm sure we should be able to delegate the ploting (and data collection) but at the very list all the matplotlib cosmetics could be done like follows:

```py
f, ax1 = plt.subplots()
ax1.plot(x, 100 * ppes, '.-')
ax1.set(title='Peak Position Error', ylabel='Error (cm)', xlabel='Threshold',
        xticks=x, xticklabels=thresholds,)
```

Reply:
I did not realize that metrics are wrapped from **sklearn**.